File: .net 6.0/ListenAI.LLM/Models/Request/CompletionOptions.cs
using ListenAI.LLM.Models.Response;
using Newtonsoft.Json;

namespace ListenAI.LLM.Models; 

public class CompletionOptions {
    /// <summary>
    /// [必填] 会话和提问 (需要自行调用 AddMessage 拼接对话历史信息，数组最后是最新的用户提问)
    /// </summary>
    [JsonRequired, JsonProperty("messages")]
    public List<SparkMessage> Messages { get; set; } = new();

    /// <summary>
    /// 用户id，一个代表你的最终用户的唯一标识符 (字符串最大长度为32)
    /// </summary>
    [JsonProperty("uid")]
    public string? UserId { get; set; }

    /// <summary>
    /// 内容审核的场景策略 (可取值： default ；默认值为 default)
    /// </summary>
    [JsonIgnore]
    public SparkOptionAuditing? Auditing { get; set; }

    [JsonProperty("auditing")]
    internal string? AuditingRaw => Auditing?.Value?.ToString();

    /// <summary>
    /// 需要使用的领域 (可取值： general ；默认值为 general)
    /// </summary>
    [JsonIgnore]
    public SparkOptionDomain? Domain { get; set; }

    [JsonProperty("domain")]
    internal string? DomainRaw => Domain?.Value?.ToString();

    [JsonIgnore]
    private decimal? _randomThreshold;

    /// <summary>
    /// 核采样阈值 (最小值是0，最大值是1，默认值为0)
    /// </summary>
    [JsonProperty("random_threshold")]
    public decimal? RandomThreshold {
        get => _randomThreshold;
        set {
            if (value is < 0.0m or > 1.0m) {
                throw new SparkException(new SparkError() {
                    Code = "FST_ERR_PARAMETER",
                    Message = "Bad Request",
                    Details = "RandomThreshold should be within range of 0-1"
                });
            }
            
            _randomThreshold = value;
        }
    }

    private int? _maxTokens;

    /// <summary>
    /// 回答的tokens的最大长度 (最小值是1，最大值是4096，默认值为1024)
    /// </summary>
    [JsonProperty("max_tokens")]
    public int? MaxTokens {
        get => _maxTokens;
        set {
            if (value is < 1 or > 4096) {
                throw new SparkException(new SparkError() {
                    Code = "FST_ERR_PARAMETER",
                    Message = "Bad Request",
                    Details = "MaxTokens should be within range of 1-4096"
                });
            }

            _maxTokens = value;
        }
    }

    /// <summary>
    /// 是否支持流式 (目前 SDK 不支持流式)
    /// </summary>
    [JsonProperty("stream")]
    public bool? Stream { get; set; }

    /// <summary>
    /// 向会话和提问数组中增加消息
    /// </summary>
    /// <param name="message">消息文本内容</param>
    /// <param name="role">角色</param>
    public void AddMessage(string message, SparkMessageRole role) {
        Messages.Add(new SparkMessage() {
            Content = message,
            Role = role
        });
    }

    /// <summary>
    /// 向会话和提问数组中增加消息
    /// </summary>
    /// <param name="message">消息</param>
    public void AddMessage(SparkMessage message) {
        Messages.Add(message);
    }

    /// <summary>
    /// 转换为 JSON
    /// </summary>
    /// <returns></returns>
    public override string ToString() {
        return JsonConvert.SerializeObject(this, new JsonSerializerSettings() {
            NullValueHandling = NullValueHandling.Ignore
        });
    }
}

File: java 17/llm/src/main/java/com/listenai/llm/models/request/CompletionOptions.java
package com.listenai.llm.models.request;

import com.alibaba.fastjson2.JSON;
import com.alibaba.fastjson2.annotation.JSONField;

import java.util.ArrayList;

public class CompletionOptions {
    @JSONField(name = "messages")
    private ArrayList<SparkMessage> messages;

    @JSONField(name = "uid")
    private String userId;

    @JSONField(name = "auditing")
    private String auditing;

    @JSONField(name = "domain")
    private String domain;

    @JSONField(name = "random_threshold")
    private float randomThreshold;

    @JSONField(name = "max_tokens")
    private Integer maxTokens;

    @JSONField(name = "stream")
    private Boolean stream;

    public CompletionOptions() {
        messages = new ArrayList<>();
        maxTokens = 1024;
    }

    public void addMessage(String message, String role) {
        SparkMessage msg = new SparkMessage(role, message);
        this.messages.add(msg);
    }

    public void addMessage(SparkMessage msg) {
        this.messages.add(msg);
    }

    public ArrayList<SparkMessage> getMessages() {
        return messages;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getAuditing() {
        return auditing;
    }

    public void setAuditing(String auditing) {
        if (auditing != "default") {
            throw new RuntimeException("Valid options for [auditing] are [default]");
        }
        this.auditing = auditing;
    }

    public String getDomain() {
        return domain;
    }

    public void setDomain(String domain) {
        if (domain != "general") {
            throw new RuntimeException("Valid options for [domain] are [general]");
        }
        this.domain = domain;
    }

    public float getRandomThreshold() {
        return randomThreshold;
    }

    public void setRandomThreshold(float randomThreshold) {
        if (randomThreshold < 0.0 || randomThreshold > 1.0) {
            throw new RuntimeException("Valid options for [random_threshold] are [0-1]");
        }
        this.randomThreshold = randomThreshold;
    }

    public Integer getMaxTokens() {
        return maxTokens;
    }

    public void setMaxTokens(Integer maxTokens) {
        if (maxTokens < 1 || maxTokens > 4096) {
            throw new RuntimeException("Valid options for [max_tokens] are [1-4096]");
        }
        this.maxTokens = maxTokens;
    }

    public Boolean getStream() {
        return stream;
    }

    public void setStream(Boolean stream) {
        if (stream) {
            throw new RuntimeException("Valid options for [stream] are [false]");
        }
        this.stream = false;
    }

    @Override
    public String toString() {
        return JSON.toJSONString(this);
    }
}
